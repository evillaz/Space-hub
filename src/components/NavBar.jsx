import { NavLink } from 'react-router-dom';
import spaceLogo from './Images/planet.png';
import './styles/NavBar.css';

const NavBar = () => (
  <>
    <div className="nav-bar">
      <NavLink to="/" style={{ color: '#000' }}>
        <div className="logo">
          <img src={spaceLogo} className="image" alt="Space Logo" />
          <h1>Space Travelers Hub</h1>
        </div>
      </NavLink>
      <nav className="nav-items">
        <ul className="menu-list">
          <li className="list-item"><NavLink to="/">Rockets</NavLink></li>
          <li className="list-item"><NavLink to="missions">Missions</NavLink></li>
          <li className="list-item"><NavLink to="profile">My Profile</NavLink></li>
        </ul>
      </nav>
    </div>
    <hr />
  </>
);

export default NavBar;
