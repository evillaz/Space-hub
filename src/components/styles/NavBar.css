.nav-bar {
    display: flex;
    margin: 20px auto 0 20px;
    align-items: center;
    justify-content: space-between;
}

.logo {
    display: flex;
    align-items: center;
    justify-content: center;
    margin: 0 20px;
    text-decoration: none;
}

.image {
    width: 50px;
    height: 50px;
    text-decoration: none; 
}

h1 {
    margin-left: 20px;
    font-size: 30px;
    font-weight: bold;
    text-decoration: none;
}

.menu-list{
    display: flex;
    margin: 0 20px;
    align-items: center;
    justify-content: center;
    text-decoration: none;
}
 .list-item {
    list-style: none;
    margin: 0 10px;
    font-size: 20px;
    font-weight: bold;
    color: rgb(6, 6, 204);
    text-decoration: none;
}

.list-item:hover {
        color: rgb(3, 3, 136);
}

.list-tem NavLink {
    text-decoration: none;
}

hr {
    width: 100%;
    margin: 10px 0 0 0;
}