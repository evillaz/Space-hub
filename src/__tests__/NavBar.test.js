import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import NavBar from '../components/NavBar';

describe('NavBar', () => {
  it('Testing of NavBar.jsx', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
