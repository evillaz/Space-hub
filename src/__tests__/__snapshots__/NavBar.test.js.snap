// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['NavBar Testing of NavBar.jsx 1'] = `
<DocumentFragment>
  <div
    class="nav-bar"
  >
    <a
      aria-current="page"
      class="nav-Header active"
      href="/"
      style="color: rgb(0, 0, 0);"
    >
      <div
        class="logo"
      >
        <img
          alt="Space Logo"
          src="planet.png"
        />
        <h1>
          Space Travelers Hub
        </h1>
      </div>
    </a>
    <nav
      class="nav-items"
    >
      <ul>
        <li
          class="list-item"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
            style="color: rgb(0, 0, 0);"
          >
            Rockets
          </a>
        </li>
        <li
          class="list-item"
        >
          <a
            class=""
            href="/missions"
            style="color: rgb(0, 0, 0);"
          >
            Missions
          </a>
        </li>
        <li
          class="list-item"
        >
          <a
            class=""
            href="/profile"
            style="color: rgb(0, 0, 0);"
          >
            My Profile
          </a>
        </li>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;
